incubator/catalog-apicurio/src/main/java/io/aklivity/zilla/runtime/catalog/apicurio/internal/ApicurioCatalogHandler.java

```diff
@@ -175,12 +192,13 @@ public int decode(
         int valLength = -1;
         if (data.getByte(index) == MAGIC_BYTE)
         {
-            progress += BitUtil.SIZE_OF_BYTE;
-            schemaId = data.getInt(index + progress, ByteOrder.BIG_ENDIAN);
-            progress += BitUtil.SIZE_OF_INT;
+            progress += SIZE_OF_BYTE;
+            schemaId = idEncoding.equals(DEFAULT_ID_ENCODING) ? (int) data.getLong(index + progress, ByteOrder.BIG_ENDIAN) :
+                data.getInt(index + progress, ByteOrder.BIG_ENDIAN);
+            progress += idEncoding.equals(DEFAULT_ID_ENCODING) ? SIZE_OF_LONG : BitUtil.SIZE_OF_INT;
```

Comment:
Let's avoid checking idEncoding on every call to decode when it is already known via options when handler is constructed.